test/logger-errors.js

```diff
@@ -0,0 +1,267 @@
+  })
+
+  t.test('timestamp is out of range', (tt) => {
+    logger.info('my log line', {
+      timestamp: startTime + configs.MS_IN_A_DAY
+    }, (err, res) => {
+      tt.error(err, 'Error was not returned')
+      tt.deepEqual(res.lines, ['my log line'], 'Log entry was accepted')
+      const data = JSON.parse(res.body.config.data)
+      const payload = data.ls[0]
+      tt.ok(payload.timestamp > startTime, 'Bad time was replaced with a good time')
+      tt.end()
+    })
+  })
+})
+
+test('.log() truncates long opts strings', (t) => {
+  const logger = new Logger(apiKey, createOptions())
+
+  t.on('end', async () => {
+    nock.cleanAll()
+  })
+
+  nock(logger._url)
+    .post('/', () => { return true })
+    .query(() => { return true })
+    .reply(200, 'Ingester response')
+
+  logger.log('log line', 'x'.repeat(configs.MAX_INPUT_LENGTH + 1), (err, res) => {
+    // This test should actually throw due to an invalid level string.  Fix later.
+    t.error(err, 'No error')
+    t.end()
+  })
+})
+
+test('.log() rejects invalid `opts` data type', (t) => {
+  const logger = new Logger(apiKey, createOptions())
+
+  t.on('end', async () => {
+    nock.cleanAll()
+  })
+
+  nock(logger._url)
+    .post('/', () => { return true })
+    .query(() => { return true })
+    .reply(200, 'Ingester response')
+
+  logger.log('log line', 12345, (err, res) => {
+    t.type(err, Error, 'Error was returned')
+    t.equal(
+      err.message
+    , 'options parameter must be a level (string), or object'
+    , 'Error message is correct'
+    )
+    t.deepEqual(err.meta, {opts: 12345}, 'Meta is set on the error')
+    t.end()
+  })
+})
+
+test('_bufferLog() error handling', async (t) => {
+  t.test('Ignores null message', (tt) => {
+    const logger = new Logger(apiKey, createOptions())
+    logger._bufferLog(null, (err, body) => {
+      tt.error(err, 'No error')
+      tt.deepEqual(body, undefined, 'No error, but nothing was done either')
+      tt.end()
+    })
+  })
+
+  t.test('Ignores empty object', (tt) => {
+    const logger = new Logger(apiKey, createOptions())
+    logger._bufferLog({}, (err, body) => {
+      tt.error(err, 'No error')
+      tt.deepEqual(body, undefined, 'No error, but nothing was done either')
+      tt.end()
+    })
+  })
+
+  t.test('Truncates messages that are too long', (tt) => {
+    tt.on('end', async () => {
+      nock.cleanAll()
+    })
+    const logger = new Logger(apiKey, createOptions())
+
+    nock(logger._url)
+      .post('/', () => { return true })
+      .query(() => { return true })
+      .reply(200, 'Ingester response')
+
+    const longData = 'x'.repeat(configs.MAX_LINE_LENGTH)
+    logger._bufferLog({line: longData + 'EXTRA'}, (err, res) => {
+      tt.error(err, 'No error')
+      const expected = [longData + ' (cut off, too long...)']
+      tt.deepEqual(res.lines, expected, 'Long line was truncated')
+      tt.end()
+    })
+  })
+
+  t.test('Ignores messages whose byte length is too long', (tt) => {
```

Comment:
For clarify can this test name be updated to show that the line's too long for the buffer, not too long as a line? Contrasting with the test right above it where the line is too long so the line's truncated instead of dropped.

Reply:
Sure, frankly, I was somewhat confused as to what the code was doing there.  I'll have another look.

Reply:
fixed